src/benchmark.cc

```diff
@@ -807,19 +811,35 @@ void RunMatchingBenchmarks(const std::string& spec,
   }
 }
 
+std::unique_ptr<BenchmarkReporter> GetDefaultReporter() {
+  typedef std::unique_ptr<BenchmarkReporter> PtrType;
+  if (FLAGS_benchmark_format == "tabular") {
+    return PtrType(new ConsoleReporter);
+  }
+  else if (FLAGS_benchmark_format == "json") {
+    return PtrType(new JSONReporter);
+  } else {
+    std::cerr << "Unexpected format: '" << FLAGS_benchmark_format << "'\n";
+    std::exit(1);
+  }
+}
+
 } // end namespace internal
 
 void RunSpecifiedBenchmarks() {
   RunSpecifiedBenchmarks(nullptr);
 }
 
-void RunSpecifiedBenchmarks(BenchmarkReporter* provided_reporter) {
+void RunSpecifiedBenchmarks(BenchmarkReporter* reporter) {
   std::string spec = FLAGS_benchmark_filter;
   if (spec.empty() || spec == "all")
     spec = ".";  // Regexp that matches all benchmarks
-  ConsoleReporter default_reporter;
-  BenchmarkReporter* reporter = provided_reporter ? provided_reporter
-                                                  : &default_reporter;
+
+  std::unique_ptr<BenchmarkReporter> default_reporter;
```

Comment:
this could be inside the if statement, right? and there's no point in it being a unique_ptr in this case as ownership is not passed.

if RunMatchingBenchmarks took a unique_ptr then this might be useful...